File: backend-nestjs/src/modules/user-management/application/commands/update-user.command.ts
import { UserRole } from '../../domain/value-objects/user-role';

export class UpdateUserCommand {
  constructor(
    public readonly id: string,
    public readonly email?: string,
    public readonly username?: string,
    public readonly firstName?: string,
    public readonly lastName?: string,
    public readonly role?: UserRole,
    public readonly departmentId?: string | null,
    public readonly isActive?: boolean,
  ) {}
}

File: backend/dotnet/clean-architecture/full/src/Core/App.Template.Application/Features/UserManagement/Commands/UpdateUser/UpdateUserCommand.cs
using AppTemplate.Application.DTOs;
using MediatR;

namespace AppTemplate.Application.Features.UserManagement.Commands.UpdateUser;

public record UpdateUserCommand : IRequest<UserDto>
{
    public long Id { get; init; }
    public string? Email { get; init; }
    public string? FirstName { get; init; }
    public string? LastName { get; init; }
    public string? Role { get; init; }
    public long? DepartmentId { get; init; }
    public bool? IsActive { get; init; }
}
